airflow-core/src/airflow/dag_processing/collection.py

```diff
@@ -637,8 +637,8 @@ def _get_dag_assets(
     inlets: bool = True,
     outlets: bool = True,
 ) -> Iterable[tuple[str, AssetT]]:
-    for task in dag.data["dag"]["tasks"]:
-        task = task[Encoding.VAR]
+    for task_raw in dag.data["dag"]["tasks"]:
```

Comment:
```suggestion
    for raw_task in dag.data["dag"]["tasks"]:
```

`raw_task` fits my taste better, but non-blocking